bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/validation/SitemapValidator.xtend

```diff
@@ -83,4 +85,34 @@ class SitemapValidator extends AbstractSitemapValidator {
 		}
 	}
 
+	@Check
+	def void checkSetpoints(Setpoint sp) {
+	    if(sp.item === null)
+	    {
+	        error("Setpoint without an item defined",
+                    SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT));
+            return
```

Comment:
IIRC you can have multiple error markers for the same feature - so do you really want to return early?

Reply:
Thanks, I was not aware of that, this little exercise was my first on xtext validation ;)